src/Symfony/Component/Translation/Writer/TranslationWriterInterface.php

```diff
@@ -0,0 +1,34 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Translation\Writer;
+
+use Symfony\Component\Translation\Exception\InvalidArgumentException;
+use Symfony\Component\Translation\MessageCatalogue;
+
+/**
+ * TranslationWriter writes translation messages.
+ *
+ * @author Michel Salib <michelsalib@hotmail.com>
+ */
+interface TranslationWriterInterface
+{
+    /**
+     * Writes translation from the catalogue according to the selected format.
+     *
+     * @param MessageCatalogue $catalogue The message catalogue to dump
+     * @param string           $format    The format to use to dump the messages
+     * @param array            $options   Options that are passed to the dumper
+     *
+     * @throws InvalidArgumentException
+     */
+    public function writeTranslations(MessageCatalogue $catalogue, $format, $options = array());
```

Comment:
I guess we can't change this now, but according to [Symfony naming conventions](https://symfony.com/doc/current/contributing/code/conventions.html), *"When an object has a "main" many relation with related "things" (objects, parameters, ...), the method names are normalized"*. So, this method should probably be called `write()` instead of `writeTranslations()`.